From 2ff40700482033ee617a444534d86b183668d46c Mon Sep 17 00:00:00 2001
From: Ramkumar Radhakrishnan <ramkumar@codeaurora.org>
Date: Wed, 7 Nov 2012 11:44:59 -0800
Subject: [PATCH 1/2] libhardware: Add allocSize hook function.

Add allocSize hook function to support the custom buffer sizes

Change-Id: I0d55195cbc02af1001d6164ce8fd55380750c029
---
 include/hardware/gralloc.h | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/include/hardware/gralloc.h b/include/hardware/gralloc.h
index 07ac029..196622e 100644
--- a/include/hardware/gralloc.h
+++ b/include/hardware/gralloc.h
@@ -343,6 +343,19 @@ typedef struct gralloc_module_t {
 typedef struct alloc_device_t {
     struct hw_device_t common;
 
+    /*
+     * (*allocSize)() Allocates a buffer in graphic memory with the requested
+     * bufferSize parameter and returns a buffer_handle_t and the stride in
+     * pixels to allow the implementation to satisfy hardware constraints on
+     * the width of a pixmap (eg: it may have to be multiple of 8 pixels).
+     * The CALLER TAKES OWNERSHIP of the buffer_handle_t.
+     *
+     * Returns 0 on success or -errno on error.
+     */
+    int (*allocSize)(struct alloc_device_t* dev,
+            int w, int h, int format, int usage,
+            buffer_handle_t* handle, int* stride, int bufferSize);
+
     /* 
      * (*alloc)() Allocates a buffer in graphic memory with the requested
      * parameters and returns a buffer_handle_t and the stride in pixels to
-- 
2.1.4

